hypertrace-core-graphql-attribute-scope/src/main/java/org/hypertrace/core/graphql/atttributes/scopes/HypertraceCoreAttributeScope.java

```diff
@@ -3,6 +3,22 @@
 import org.hypertrace.core.graphql.common.schema.attributes.AttributeScope;
 
 enum HypertraceCoreAttributeScope implements AttributeScope {
-  TRACE,
-  SPAN;
+  TRACE(HypertraceCoreAttributeScopeString.TRACE),
```

Comment:
Shouldn't we rather remove this enum? Or, are you still keeping it necessary to make changes for scope addition in hypertrace-core?

Reply:
In the current design (because of the enum, and the inability to extend it) there's a separate enum in each GQL app. The code interacts with it through the interface `AttributeScope`, with different implementations bound in each application.

So with that context, the piece to remove here is `AttributeScope`. Because that is part of the public API, it can't be removed in a single PR - it must be deprecated. This PR removes its internal usage, deprecates it in response fields (i.e `AttributeMetadata`), and provides a string alternative. The corresponding HT PR does the same in that repo. The work still to be done is to accept a string alternatives when scopes are provided in arguments, and then eventually to remove the enums like this entirely.